# **Data Preparation**:

In [4]:
#importing libraries 
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.metrics import accuracy_score, classification_report
import pandas as pd
from sklearn.metrics import accuracy_score
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split

In [5]:
#loading  and previewing rainfall data sets:
rainfal=pd.read_csv("kenya_rainfall_2022_2024.csv")
rainfal.head()

time  precipitation_sum (mm)  Unnamed: 2  Unnamed: 3  Unnamed: 4  \
0  01/01/2022                     0.0         NaN         NaN         NaN   
1  02/01/2022                     0.0         NaN         NaN         NaN   
2  03/01/2022                     0.0         NaN         NaN         NaN   
3  04/01/2022                     0.0         NaN         NaN         NaN   
4  05/01/2022                     0.0         NaN         NaN         NaN   

   Unnamed: 5     town  
0         NaN  Nairobi  
1         NaN  Nairobi  
2         NaN  Nairobi  
3         NaN  Nairobi  
4         NaN  Nairobi

In [6]:
#loading and previewing wfp food prices
foodprices=pd.read_csv('wfp_food_prices_ken.csv')
foodprices.head()

date      admin1      admin2            market         market_id  \
0       #date  #adm1+name  #adm2+name  #loc+market+name  #loc+market+code   
1  2006-01-15       Coast     Mombasa           Mombasa               191   
2  2006-01-15     Eastern       Kitui             Kitui               187   
3  2006-01-15     Eastern       Kitui             Kitui               187   
4  2006-01-15     Eastern       Kitui             Kitui               187   

   latitude longitude            category         commodity commodity_id  \
0  #geo+lat  #geo+lon          #item+type        #item+name   #item+code   
1     -4.05     39.67     pulses and nuts             Beans           50   
2     -1.37     38.02  cereals and tubers  Potatoes (Irish)          148   
3     -1.37     38.02  cereals and tubers           Sorghum           65   
4     -1.37     38.02     pulses and nuts       Beans (dry)          262   

         unit         priceflag         pricetype        currency    price  \
0  #item+unit  #item+price+flag  #item+price+type  #currency+code   #value   
1          KG            actual         Wholesale             KES    33.63   
2       50 KG            actual         Wholesale             KES  1249.99   
3       90 KG            actual         Wholesale             KES     1800   
4          KG            actual            Retail             KES       39   

     usdprice  
0  #value+usd  
1        0.47  
2       17.39  
3       25.03  
4        0.54

In [7]:
#loading and previewing cities data:
towns=pd.read_csv('cities_latitude_and_longitudes.csv')
towns.head()

city     lat      lng country iso2    admin_name  capital  population  \
0  Nairobi -1.2864  36.8172   Kenya   KE  Nairobi City  primary   5545000.0   
1  Mombasa -4.0500  39.6667   Kenya   KE       Mombasa    admin   1208333.0   
2   Nakuru -0.3000  36.0667   Kenya   KE        Nakuru    admin    570674.0   
3  Eldoret  0.5167  35.2833   Kenya   KE   Uasin Gishu    admin    475716.0   
4   Kisumu -0.0833  34.7667   Kenya   KE        Kisumu    admin    409928.0   

   population_proper  
0          5545000.0  
1          1200000.0  
2           307990.0  
3           289380.0  
4           409928.0

In [8]:
#previewing foodprices columns
list(foodprices.columns)

['date',
 'admin1',
 'admin2',
 'market',
 'market_id',
 'latitude',
 'longitude',
 'category',
 'commodity',
 'commodity_id',
 'unit',
 'priceflag',
 'pricetype',
 'currency',
 'price',
 'usdprice']

In [9]:
#previewing rainfal columns
list(rainfal.columns)

['time',
 'precipitation_sum (mm)',
 'Unnamed: 2',
 'Unnamed: 3',
 'Unnamed: 4',
 'Unnamed: 5',
 'town']

In [10]:
foodprices.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12783 entries, 0 to 12782
Data columns (total 16 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   date          12783 non-null  object
 1   admin1        12749 non-null  object
 2   admin2        12749 non-null  object
 3   market        12783 non-null  object
 4   market_id     12783 non-null  object
 5   latitude      12749 non-null  object
 6   longitude     12749 non-null  object
 7   category      12783 non-null  object
 8   commodity     12783 non-null  object
 9   commodity_id  12783 non-null  object
 10  unit          12783 non-null  object
 11  priceflag     12783 non-null  object
 12  pricetype     12783 non-null  object
 13  currency      12783 non-null  object
 14  price         12783 non-null  object
 15  usdprice      12783 non-null  object
dtypes: object(16)
memory usage: 1.6+ MB


Food price data has 12783 rows.
columns such as date price are of string data types.

In [11]:
#Checking for null values:
foodprices.isnull().sum()

date             0
admin1          34
admin2          34
market           0
market_id        0
latitude        34
longitude       34
category         0
commodity        0
commodity_id     0
unit             0
priceflag        0
pricetype        0
currency         0
price            0
usdprice         0
dtype: int64

Food prices has a total of 34 rows with null values under the region, subregion,latitude and longitude columns.

In [12]:
rainfal.isnull().sum()

time                         61
precipitation_sum (mm)       61
Unnamed: 2                69109
Unnamed: 3                69109
Unnamed: 4                69109
Unnamed: 5                69109
town                         61
dtype: int64

Unnamed columns in rainfall dataset have 69109 null values while time and precipitation columsn have 61 missing values

In [13]:
rainfal.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69109 entries, 0 to 69108
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   time                    69048 non-null  object 
 1   precipitation_sum (mm)  69048 non-null  float64
 2   Unnamed: 2              0 non-null      float64
 3   Unnamed: 3              0 non-null      float64
 4   Unnamed: 4              0 non-null      float64
 5   Unnamed: 5              0 non-null      float64
 6   town                    69048 non-null  object 
dtypes: float64(5), object(2)
memory usage: 3.7+ MB


Rainfal data has 69048 rows. Time column has string data type.
The data has 4 columns with null values 